ui/src/pages/Tags/Edit/index.tsx

```diff
@@ -85,6 +82,18 @@ const Index = () => {
     when: contentChanged,
   });
 
+  useEffect(() => {
+    setFormData({
+      ...formData,
+      displayName: { ...formData.displayName, value: data?.display_name || '' },
+      slugName: { ...formData.slugName, value: data?.slug_name || '' },
+      description: {
+        ...formData.description,
+        value: data?.original_text || '',
+      },
+    });
```

Comment:
You may have overlooked immData. immData is used to record the initial value. When the user changes the content and leaves the page, there will be a prompt; therefore, you need to add an assignment operation to immData here;

Reply:
Thanks @shuashuai , updated. There's one thing makes me confused that for below codes.
```
    useEffect(() => {
    ....
    if (!display_name || !slug_name || !original_text) {
      return;
    }
    ....
    }
```
Seems it'll never go into this condition branch.

Reply:
> Thanks @shuashuai , updated. There's one thing makes me confused that for below codes.
> 
> ```
>     useEffect(() => {
>     ....
>     if (!display_name || !slug_name || !original_text) {
>       return;
>     }
>     ....
>     }
> ```
> 
> Seems it'll never go into this condition branch.

Yes, you can remove this condition now! In addition, for the previous assignment operation, you can simplify the code and use an object to assign the value. Try to reduce the number of lines of code! Thank you for your thoughts and suggestions！

Reply:
resolved. please help to review and let me know if any comment. thanks!